### PR TITLE
Implement Freerunning mechanic (ready for review)

### PR DESCRIPTION
**Pull request by theelk801**
I should write a test for this but otherwise it should be good to go.

### CODE REVIEW

File: Mage.Sets/src/mage/cards/c/ChainAssassination.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        boolean flag = MorbidCondition.instance.apply(game, source);
+        Permanent permanent = game.getPermanent(getTargetPointer().getFirst(game, source));
+        if (permanent == null) {
+            return false;
+        }
+        permanent.destroy(source, game);
+        if (!flag) {
+            return true;
+        }
+        Player player = game.getPlayer(source.getControllerId());
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/c/ChainAssassination.java:71]: process action in between?

File: Mage.Tests/src/test/java/org/mage/test/cards/abilities/keywords/FreerunningTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        addCard(Zone.HAND, playerA, vision);
+
+        castSpell(1, PhaseStep.POSTCOMBAT_MAIN, playerA, vision);
+
+        setStrictChooseMode(true);
+        setStopAt(1, PhaseStep.END_TURN);
+        execute();
+
+        assertHandCount(playerA, 3);
+        assertTapped("Island", true);
```
Comment (xenohedron) [inline Mage.Tests/src/test/java/org/mage/test/cards/abilities/keywords/FreerunningTest.java:27]: more robust to use assertTappedCount and check that all five Islands are tapped

also worth dealing combat damage with a non-Assassin non-commander

**Review by xenohedron — APPROVED** [review]

File: Mage.Sets/src/mage/cards/c/ChainAssassination.java (hunk 1 above)
Comment (theelk801) [reply Mage.Sets/src/mage/cards/c/ChainAssassination.java:71]: I'm not sure I see the point here

File: Mage.Tests/src/test/java/org/mage/test/cards/abilities/keywords/FreerunningTest.java (hunk 2 above)
Comment (theelk801) [reply Mage.Tests/src/test/java/org/mage/test/cards/abilities/keywords/FreerunningTest.java]: good call

File: Mage.Sets/src/mage/cards/c/ChainAssassination.java (hunk 1 above)
Comment (xenohedron) [reply Mage.Sets/src/mage/cards/c/ChainAssassination.java:71]: the destroyed creature shouldn't be able to trigger off you drawing a card e.g. consecrated sphinx

**Pull request merged**